layers/+lang/c-c++/README.org

```diff
@@ -9,197 +9,201 @@
+=c-c++-lsp-enable-semantic-highlight= to non-nil. It is possible to give a
+different color for each identifier by setting the variable to ='rainbow=.
 
-#+BEGIN_SRC emacs-lisp
-  (setq-default dotspacemacs-configuration-layers
-                '((c-c++ :variables
-                         c-c++-backend 'lsp-cquery
-                         c-c++-lsp-executable (file-truename "~/bin/cquery/or/ccls"))))
-#+END_SRC
+When semantic highlighting is enabled the method used to highlight the text
+relies on the =font-lock= mechanism of Emacs which is the fastest. More
+accurate but also slower method is to use the overlays. To use the overlays set
+the layer variable =c-c++-lsp-semantic-highlight-method= to ='overlay=.
 
-***** Semantic highlighting
-Semantic highlighting is disabled by default. To enable, set the
-=c-c++-lsp-sem-highlight-method= variable to either ='font-lock= or ='overlay=.
-To enable the rainbow semantic highlighting colour theme, set
-=c-c++-lsp-sem-highlight-rainbow= to =t=.
+***** Cache directory (absolute or relative)
+Currently only available for =lsp-ccls= and =lps-cquery=.
 
-***** Additional configuration options
-Both lsp backends are configured to store their index cache in a subdirectory of
+All LSP backends are configured to store their index cache in a subdirectory of
 =.emacs.d/cache=. This can be overridden by specifying an explicit
 =c-c++-lsp-cache-dir=. Setting this value to a relative path will cause the
-index cache to be placed in a subdirectory of your project root.
+index cache to be placed in a subdirectory of your current project root.
 
-There are other initialization options such as the number of indexer threads,
-cache serialization format. They have good default values. See =config.el= of
-the layer and the backends' respective homepages for more info.
-- [[https://github.com/cquery-project/cquery/wiki/Emacs][Emacs section of =cquery= wiki]]
+***** Additional configuration options
+See the full list of configuration options you can set at:
 - [[https://github.com/MaskRay/ccls/wiki/Emacs][Emacs section of =ccls= wiki]]
+- [[https://github.com/cquery-project/cquery/wiki/Emacs][Emacs section of =cquery= wiki]]
 
 ***** Example dotspacemacs-configuration-layers entry
-#+BEGIN_SRC emacs-lisp
-  (setq-default dotspacemacs-configuration-layers
-                '((c-c++ :variables
-                         c-c++-adopt-subprojects t
-                         c-c++-backend 'lsp-ccls
-                         c-c++-lsp-executable (file-truename "~/dev/cpp/ccls/Release/ccls")
-                         c-c++-lsp-sem-highlight-rainbow t)))
-#+END_SRC
+ #+BEGIN_SRC emacs-lisp
+   (setq-default dotspacemacs-configuration-layers
+                 '((c-c++ :variables
+                          c-c++-adopt-subprojects t
+                          c-c++-backend 'lsp-ccls
+                          c-c++-lsp-executable (file-truename "~/dev/cpp/ccls/Release/ccls")
```

Comment:
This doesn't work. The readme should probably mention somewhere that you have to set `lsp-clients-clangd-executable ` for clangd, I am not sure about the other backends.

Reply:
I use clangd and I didn't have to set `lsp-clients-clangd-executable`. Are you sure your clangd is available on the PATH? What does `M-: (executable-find "clangd")` returns?

Reply:
maybe this wasn't clear: If clangd is not in your path you have to set `lsp-clients-clangd-executable`, `c-c++-lsp-executable` does **not** work for this purpose. `clangd` is relatively new and people with older distros will have to download clang manually and put it somewhere in order to be able to use `clangd`.

Reply:
Added a section to the README.org titled `Path to LSP server executables`